app/main/plugins/plugins.js

```diff
@@ -367,28 +346,33 @@ exports.queryHelper = (plugin, keyword) =>
  * @return {Promise} - Resolves to the rendered html string
  */
 exports.retrieveItemDetails = (item, plugin) =>
-  new Promise(resolve => {
+  new Promise((resolve, reject) => {
     // retrieve the rendered content
-    let type = 'html';
-    let content = '';
-    if (plugin && plugin.details) {
-      if (plugin.details.type) {
-        type = plugin.details.type;
-      }
-      if (plugin.details.render) {
-        if (typeof plugin.details.render === 'function') {
-          content = plugin.details.render(item);
-        } else {
-          content = plugin.details.render;
-        }
-      }
+    if (!plugin || !plugin.details) {
+      reject(new Error(`Missing details for ${plugin.name}`));
+      return;
     }
-    Promise.resolve(content).then(res => {
-      let html = res;
-      if (type === 'md') {
-        const md = new MarkdownIt();
-        html = md.render(res);
+    const type = plugin.details.type || 'html';
+    const content = resolveByCondition(
+      Boolean(plugin.details.render),
+      () =>
+        typeof plugin.details.render === 'function'
+          ? plugin.details.render(item)
+          : plugin.details.render,
+      () => ''
+    );
+    switch (type) {
+      case 'md': {
+        const mdRenderer = new MarkdownIt();
+        resolve(mdRenderer.render(content));
+        break;
       }
-      resolve(html);
-    });
+      case 'html':
+        resolve(content);
+        break;
+      default:
+        reject(
```

Comment:
I realise that this can be a problem for the external plugins not implementing a `type`.
Although I wouldn't want to fail silently, since this is something that is rendered via `dangerouslySetHTML`